src/clock.rs

```diff
@@ -24,12 +25,29 @@ impl Clocks {
         cpu_freq: 390000000
 */
         Self {
-            cpu: Hertz(403_000_000)
+            cpu: Hertz(403_000_000),
+            apb0: Hertz(195_000_000),
         }
     }
 
     /// Returns CPU frequency
     pub fn cpu(&self) -> Hertz {
         Hertz(self.cpu.0)
     }
+
+    // UART clocks: these equal APB0 clock *if* the individual peripheral clock is enabled
+    /// Returns UART1 frequency
+    pub fn uart1(&self) -> Hertz {
+        self.apb0
```

Comment:
I hope it's better to have just apb0() function here

Reply:
Yes, it depends on whether we want to return 0 if the clock is disabled.

Reply:
Which one? If `apb0` clock is disabled we want to return 0 here. If just `uart1` clock is disabled, this should not be reflected in `Clocks`, because this struct represents some clock tree state that is not subject to change. I think that we want enabling/disabling peripheral clocks after the clock tree is configured, so it's better to keep this information out of `Clocks` struct.

Reply:
Yeah I meant the UART1 clock. Hadn't even thought about disabling the APB0 clock itself, Seems the SDK enables the APB0 as by-effect of enabling the clock of any of the peripherals.
Agree it should probably be possible to switch them on/off after the initial setup for power management reasons.

Reply:
Hmm, thinking of it, the same is true for the frequencies, an application might want to clock down the CPU when the device is idle. I'm not sure how much clock state can be considered immutable, unless it's only a representation of the clock tree circuit in hardware without storing the specific divisors/selectors/frequencies.

I think that's what confused me. Another reasoning was whether it was the UART driver's responsibility to know it is connected to APB0: it's slightly more general to ask the clocks for its own clock, not that it matters as long as it's only part of one SoC.

Anyhow I'm fine with changing this to one `apb0()` method.

Reply:
Yes, this `Clocks` solution is not perfect. It does not support low-power modes and clock reconfiguration: we need better solution for this. I'd like to wait for it to appear for STM32L families first.

As for `APB0`: it's common to have this knowledge in HAL crate like [this](https://github.com/stm32-rs/stm32f1xx-hal/blob/8938824104b3fc435f01ea806768024215f411ba/src/serial.rs#L328).